src/Native/Unix/System.Native/pal_networking.cpp

```diff
@@ -2071,6 +2071,51 @@ extern "C" Error SystemNative_GetSockOpt(
 
     int fd = ToFileDescriptor(socket);
 
+    //
+    // Handle some special cases for compatibility with Windows
+    //
+    if (socketOptionLevel == PAL_SOL_SOCKET)
+    {
+        if (socketOptionName == PAL_SO_EXCLUSIVEADDRUSE)
+        {
+            //
+            // SO_EXCLUSIVEADDRUSE makes Windows behave like Unix platforms do WRT the SO_REUSEADDR option.
+            // So, for non-Windows platforms, we act as if SO_EXCLUSIVEADDRUSE is always enabled.
+            //
+            if (*optionLen != sizeof(int32_t))
+            {
+                return PAL_EINVAL;
+            }
+
+            *reinterpret_cast<int32_t*>(optionValue) = 1;
+            return PAL_SUCCESS;
+        }
+        else if (socketOptionName == PAL_SO_REUSEADDR)
+        {
+            //
+            // On Windows, SO_REUSEADDR allows the address *and* port to be reused.  It's equivalent to 
+            // SO_REUSEADDR + SO_REUSEPORT other systems.  Se we only return "true" if both of those options are true.
+            //
+            auto optLen = static_cast<socklen_t>(*optionLen);
+
+            int err = getsockopt(fd, SOL_SOCKET, SO_REUSEADDR, optionValue, &optLen);
+
+            if (err == 0 && *reinterpret_cast<uint32_t*>(optionValue) != 0)
+            {
+                err = getsockopt(fd, SOL_SOCKET, SO_REUSEPORT, optionValue, &optLen);
```

Comment:
As pointed out in the design discussion, this could be confusing: what if on *NIX I only want to set `SO_REUSEPORT` but not `SO_REUSEADDR`? 
Is that a a valid *NIX scenario? If yes, is there a way to achieve that somehow?